Central source check

nkhadka21 opened the pull request:
This PR implements a method to check if an image contains a bright source within the central box of a specified dimension. This is used in the LSST Science Pipeline source injection task. It ensures that our injected lens does not overlap with an existing object.

sibirrer reviewed (commented):
Overall looks good. I am a bit worried about the size of the definition that extends further. Perhaps you can write sub-definitions to make it more structured? Or at least add some more comments in the code? I am worried for future developments that it is hard to change and read the code

nkhadka21 commented:
> Overall looks good. I am a bit worried about the size of the definition that extends further. Perhaps you can write sub-definitions to make it more structured? Or at least add some more comments in the code? I am worried for future developments that it is hard to change and read the code

Hi @sibirrer , I refactored this function! It is now split into three separate functions.

sibirrer reviewed (approved):
Thank you very much! Looks good (pending tests)
